security/src/main/java/com/fincity/security/service/ClientService.java

```diff
@@ -525,6 +526,10 @@ private Mono<Tuple2<User, String>> registerUser(ClientRegistrationRequest reques
 		return this.userService.createForRegistration(user)
 		        .map(e -> Tuples.of(e, finPassword));
 	}
+	
+	public Mono<List<Package>> fetchPackages(ULong clientId) {
+		return this.dao.getPackagesAvailableForClient(clientId);
```

Comment:
Need to check if the logged in user is a system or not. Should have client read access and package read access. Should either for self or managing client. Please perform all these checks and do the same for the other service methods too.